### ISSUE TITLE
desispec.pipeline.plan.graph_path* shouldn't replicate io.findfile functionality

### ISSUE THREAD

**Issue by sbailey** [desihub/desispec#242]
`desispec.pipeline.plan.graph_path*` functions have a lot of code like:

```
    dir = os.path.join(proddir, 'exposures', night, expid) 
    path = os.path.join(dir, "frame-{}-{}.fits".format(cam, expid)) 
```

This replicates the functionality in `io.findfile()`, which is supposed to be the canonical way of mapping a file type plus camera, expid, night, etc. into a location on disk.

Update `desispec.pipeline.plan.graph_path*` to use `io.findfile()` instead.  If `io.findfile` has missing functionality, add it there.

**Comment by sbailey** [comment]
desispec.pipeline.run.qa_path() is another area that should tie into desispec.io.findfile to get the canonical name and location.

**Comment by tskisner** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Pipeline Improvements](https://trello.com/c/sCuBELo3/4-pipeline-improvements)

**Comment by tskisner** [comment]
Rather than fix this code and then change that same code as part of #292, I am going to leave this open until that is finished.

**Comment by tskisner** [comment]
Closed by 1468b777f3a4eb2ad565cb10fb6fc1d3044ceaf4
